Patch to add away message to nick menu,
by Christopher Aillon <caillon@redhat.com>

--- ../xchat-2.4.2-orig/src/fe-gtk/menu.c	2005-03-14 23:44:12.000000000 -0500
+++ src/fe-gtk/menu.c	2005-03-21 15:18:46.000000000 -0500
@@ -450,12 +450,13 @@ static char *str_copy = 0;		/* for all p
 
 void
 menu_nickmenu (session *sess, GdkEventButton *event, char *nick, int num_sel)
 {
 	char buf[256];
 	struct User *user;
+	struct away_msg *away;
 	GtkWidget *wid, *submenu, *menu = gtk_menu_new ();
 
 	if (str_copy)
 		free (str_copy);
 	str_copy = strdup (nick);
 
@@ -487,12 +488,20 @@ menu_nickmenu (session *sess, GdkEventBu
 			menu_quick_item (0, buf, submenu, 0, 0);
 
 			snprintf (buf, sizeof (buf), _("Server: %s"),
 						user->servername ? user->servername : _("Unknown"));
 			menu_quick_item (0, buf, submenu, 0, 0);
 
+			if (user->away)
+			{
+				away = find_away_message (current_sess->server, nick);
+				snprintf (buf, sizeof (buf), _("Away Msg: %s"),
+						away && away->message ? away->message : _("Unknown"));
+				menu_quick_item (0, buf, submenu, 0, 0);
+			}
+
 			snprintf (buf, sizeof (buf), _("Last Msg: %s"),
 						user->lasttalk ? ctime (&(user->lasttalk)) : _("Unknown"));
 			if (user->lasttalk)
 				buf[strlen (buf) - 1] = 0;
 			wid = menu_quick_item (0, buf, submenu, 0, 0);
 
